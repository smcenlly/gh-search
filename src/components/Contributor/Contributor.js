import React from "react";
import i18n from "i18next";
import styled from "styled-components";
import PropTypes from "prop-types";

const ContributorItem = styled.li`
    & + & {
        margin-top: 12px;
    }
`;

class Contributor extends React.PureComponent {
    static propTypes = {
        login: PropTypes.string,
        contributions: PropTypes.number
    };
    render() {
        let { className, login, contributions } = this.props;
        return (
            <ContributorItem className={className}>
                {i18n.t("contributor.contributions", {
                    login,
                    count: contributions
                })}
            </ContributorItem>
        );
    }
}

export default Contributor;
