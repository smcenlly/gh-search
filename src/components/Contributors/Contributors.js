import React from "react";
import i18n from "i18next";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Contributor } from "../Contributor";

const ContributorsList = styled.ul`
    margin-bottom: 0;
`;

const NoContributions = styled.li`
    list-style: none;
`;

class Contributors extends React.PureComponent {
    static propTypes = {
        contributors: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.object),
            PropTypes.string
        ])
    };

    static NoContributions = NoContributions;

    render() {
        let { className, contributors } = this.props;
        return (
            <ContributorsList className={className}>
                {contributors.length > 0 ? (
                    contributors.map(({ login, contributions, id }) => {
                        return (
                            <Contributor
                                key={id}
                                login={login}
                                contributions={contributions}
                            />
                        );
                    })
                ) : (
                    <NoContributions>
                        {i18n.t("contributors.noContributors")}
                    </NoContributions>
                )}
            </ContributorsList>
        );
    }
}

export default Contributors;
