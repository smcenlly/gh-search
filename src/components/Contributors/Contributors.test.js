import React from "react";
import { shallow } from "enzyme";
import Contributors from "./Contributors";
import { Contributor } from "../Contributor";

describe("Contributors", function() {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Contributors contributors={[]} />);
    });

    it("renders without crashing", () => {
        expect(wrapper).toExist();
    });

    it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe("when provided contributors", () => {
        it("renders Contributors with correct props", () => {
            const contributors = [
                { id: 1, login: "loginName1", contributions: 100 },
                { id: 2, login: "loginName2", contributions: 200 }
            ];

            const contributorProp = {
                login: contributors[0].login,
                contributions: contributors[0].contributions
            };

            wrapper.setProps({ contributors });

            expect(wrapper).toContainMatchingElements(2, Contributor);
            expect(wrapper.find(Contributor).first()).toHaveProp(
                contributorProp
            );
        });
    });

    describe("when not provided repos", () => {
        it("renders no contributors found block", () => {
            expect(wrapper).toContainExactlyOneMatchingElement(
                Contributors.NoContributions
            );
        });
    });
});
