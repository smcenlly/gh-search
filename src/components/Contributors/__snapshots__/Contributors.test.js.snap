// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributors matches snapshot 1`] = `
<styled.ul>
  <styled.li />
</styled.ul>
`;
